### ISSUE TITLE
Should not lint on R6 cloned objects

### ISSUE THREAD

**Issue by radbasa** [Appsilon/box.linters#26]
## box.linters version

0.0.0.9006

## Source code to lint

```R
box::use(
  R6[R6Class],
)

SomeClass <- R6Class("SomeClass",   # nolint
  public = list(
    method = function() {
      "method"
    }
  )
)

s <- SomeClass$new()

t <- s$clone()
t$method()
```

## Lint command

```R
lintr::lint(code, linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters))
```

## Lint result

```
<text>:16:6: warning: [box_usage_linter] <package/module>$function does not exist.
t <- s$clone()
     ^~~~~~~
<text>:17:1: warning: [box_usage_linter] <package/module>$function does not exist.
t$method()
^~~~~~~~
```

## Related issues

- #22 

## Expected result

- [ ] Should not lint on R6 `clone()`
- [ ] Should not lint on clone R6 method call.
